### PR TITLE
Ensure literals have the right argument. Create literals with a requested precision

### PR DESCRIPTION
**Pull request by EmilyBourne**
Add type checks to Literal classes to ensure that they contain arguments of the correct type. Use the precision argument to translate precision modifiers of literals to a literal of the correct precision. E.g:
```
np.float32(2)
```
Is now translated to:
```
2.0_C_FLOAT
```
instead of:
```
a = Real(2_C_LONG_LONG, 4)
```
Casts are still printed for truncated values

**Commit summary**
- Use inheritance to set `Literal` precision and check its type
- Remove unused `sympy.Integer` inheritance from `LiteralInteger` (this has many knock-on effects as it makes it obvious when we are not using pyccel operators)
- Add type checks and conversions to `Literal` classes
- Use literal precision to translate expressions such as `np.float32(2)` directly to `2.0_C_FLOAT`
- Print floats with correct precision in fortran
- Fix `unexpected keyword argument : "precision"`
- ensure that `complex` function works when complex arguments are passed
- Simplify phrases such as `2+5j` to `LiteralComplex(2,5)`

### CODE REVIEW

File: pyccel/ast/literals.py [hunk 1]
```diff
@@ -29,59 +27,114 @@ class Literal(PyccelAstNode):
...
+            if real.precision == precision:
+                self._real_part = real
+            else:
+                self._real_part = LiteralFloat(real.args[0], precision = precision)
+        elif isinstance(real, LiteralInteger):
+            self._real_part = LiteralFloat(real.p, precision = precision)
+        elif isinstance(real, (int, float)):
+            self._real_part = LiteralFloat(real, precision = precision)
+        else:
+            raise TypeError("The real part of a LiteralComplex must be an int/float/LiteralFloat")
```
Comment (yguclu) [inline pyccel/ast/literals.py:125]: Is a `LiteralInteger` not allowed?

File: pyccel/ast/operators.py [hunk 2]
```diff
@@ -447,6 +449,14 @@ class PyccelAdd(PyccelArithmeticOperator):
     """
     _precedence = 12
 
+    def __new__(cls, arg1, arg2):
+        if isinstance(arg1, (LiteralInteger, LiteralFloat)) and \
+           isinstance(arg2, LiteralComplex) and \
+           arg2.real == LiteralFloat(0):
+            return LiteralComplex(arg1, arg2.imag)
```
Comment (yguclu) [inline pyccel/ast/operators.py:456]: As far as I understand this handles a case like `1 + 2j`, but why shouldn't it also handle `2j + 1`?

File: pyccel/ast/operators.py [hunk 3]
```diff
@@ -488,6 +498,14 @@ class PyccelMinus(PyccelArithmeticOperator):
     """
     _precedence = 12
 
+    def __new__(cls, arg1, arg2):
+        if isinstance(arg1, LiteralFloat) and \
+           isinstance(arg2, LiteralComplex) and \
+           arg2.real_part == LiteralFloat(0):
+            return LiteralComplex(arg1, -float(arg2.imag_part))
```
Comment (yguclu) [inline pyccel/ast/operators.py:505]: Above you used the properties `real` and `imag`, here you are using `real_part` and `imag_part` instead. What's the difference?

File: tests/epyccel/modules/complex_func.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    a = complex(2.8+7j,1)
+    return a
+
+def create_complex_literal__complex_float():
+    a = complex(2.8+7j,1.8)
+    return a
+
+def create_complex_literal__complex_complex():
+    a = complex(2.8+7j,1.5+22j)
+    return a
```
Comment (yguclu) [inline tests/epyccel/modules/complex_func.py:39]: Maybe throw in some negative numbers for good measure?

File: pyccel/ast/operators.py (hunk 3 above)
Comment (EmilyBourne) [reply pyccel/ast/operators.py]: `real_part` and `imag_part` are the properties of `PythonComplex` while `real` and `imag` are the properties of `LiteralComplex`. We should probably use the same for both. Which do you prefer?

Comment (yguclu) [reply pyccel/ast/operators.py]: I prefer the shortest names: `real` and `imag`

**Review by yguclu — APPROVED** [review]

**Pull request merged**